### PR TITLE
sys: Fix pthread includes to support avr-libs

### PR DESCRIPTION
**Pull request by rzr**
Missing malloc.h and clock_id_t were causing issues to build.

It was tested with this configuration:

- linux ubuntu 14.04.5
- arduino-mega2560 board
- avr-libc-1.8.0-4.1

Change-Id: I2080dfb43c75805b255f6c65a09ee2c0c0e54f6a
Signed-off-by: Philippe Coval <philippe.coval@osg.samsung.com>

### CODE REVIEW

File: sys/posix/pthread/pthread_tls.c [hunk 1]
```diff
@@ -16,10 +16,14 @@
  * @}
  */
 
-#include <malloc.h>
+
 
 #include "pthread.h"
 
+#ifndef __WITH_AVRLIBC__
```
Comment (aabadie) [inline sys/posix/pthread/pthread_tls.c]: Looks redundant with [here](https://github.com/RIOT-OS/RIOT/pull/6075/files#diff-afa8532f50bb093ed158e5af99044cc4R25)

Comment (rzr) [inline sys/posix/pthread/pthread_tls.c]: 1st #if is closing before my line isnt it ?

Comment (aabadie) [inline sys/posix/pthread/pthread_tls.c]: I meant you have `#include "pthread.h"`, line 21 and in `pthread.h` you already added:
```C
#ifndef __WITH_AVRLIBC__
#ifndef HAVE_MALLOC_H
#define HAVE_MALLOC_H 1
#endif
#endif
```
I'm also wondering if the `#ifndef HAVE_MALLOC_H` is superfluous.

**Comment by aabadie** [comment]
@rzr, thanks for applying the changes. How can I test it ?

**Comment by rzr** [comment]
You can at least try to build it this way, adapt for your OS

cat /etc/os-release
VERSION="14.04.5 LTS, Trusty Tahr"

sudo apt-get install gcc-avr gdb-avr avr-libc binutils-avr avrdude
# 1:1.8.0-4.1

dir=examples/hello-world/ ; 
make -C $dir BOARD=arduino-mega2560 
make -C $dir BOARD=arduino-mega2560 flash term

**Comment by aabadie** [comment]
Just tried with current [master](b24be6a6ae8f03fd5d70b23f49f822be7eb04219) and couldn't reproduce. It builds fine for atmega2560.
I tested with Ubuntu 16.10 and also on a VM running 14.04.5 (fresh install).

I didn't test the `flash` and `term` targets because of missing hardware but they don't seem to be related.

**Comment by aabadie** [comment]
I tried to build examples/iotivity/client but it always (avr, arm, ubuntu 14.04, 16.10) building tinycbor with the following error:
```
Submodule path 'deps/tinycbor': checked out '0a78b71322fe2f54b342c1b6ea66ce31da41383e'
port/riot/clock.c: In function 'oc_clock_time':
port/riot/clock.c:29:44: error: invalid operands to binary + (have 'xtimer_ticks64_t' and 'int')
   return (oc_clock_time_t)((xtimer_now64() + 999) / 1000);
                                            ^
port/riot/clock.c:30:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
cc1: all warnings being treated as error
```

**Comment by rzr** [comment]
Well at that time it fixed an example I don't remember which one, maybe:

make BOARD=arduino-mega2560 -C examples/dtls-echo WERROR=0 

I'll may have a look later

**Comment by rzr** [comment]
Yea I was for iotivity, I submitted a fix for review upstream

https://gerrit.iotivity.org/gerrit/16497 Declare void parameter for RIOT

If merged I can update revision for solving:
https://github.com/RIOT-OS/RIOT/issues/6241

Meanwhile you can try:

make="make PKG_URL=http://github.com/tizenteam/iotivity-constrained PKG_VERSION=sandbox/pcoval/on/master/review  BOARD=arduino-mega2560 "
list="\
 examples/iotivity_examples/client \
 examples/iotivity_examples/client_switch \
 examples/iotivity_examples/server \
 examples/iotivity_examples/br_fw \
"


for example in $list ; do
$make -C $example 
done

File: examples/iotivity_examples/client_switch/client_switch.c [hunk 2]
```diff
@@ -76,6 +76,7 @@ stop_observe(void *data)
     return DONE;
 }
 
+#ifdef BUTTON_GPIO
```
Comment (aabadie) [inline examples/iotivity_examples/client_switch/client_switch.c]: I think this should deserve another PR

**Comment by aabadie** [comment]
Your last changes in d98b1e61c9164257673dcd2ff59f67bd281f6ce0 indeed fix the build issue.

ACK when previous comment is addressed and CI green
